src/main/java/core/basesyntax/OperationStrategyImpl.java

```
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import core.interfaces.OperationHandler;
+import core.interfaces.OperationStrategy;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation,
+            OperationHandler> handlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> handlers) {

```

Constructor does not validate input: `handlers` may be null or later mutated externally. Add a null check and defensively copy (and optionally wrap with `Collections.unmodifiableMap`) to preserve immutability and fail fast on bad input. This enforces null-parameter handling called out in the checklist.